### ISSUE TITLE
CoE Compliance and Report solution error

### ISSUE THREAD

**Issue by melmgcoe** [microsoft/powerapps-tools#280]
Hello,

I'm having issues turning on the following flow as part of CoE Compliance and Report solution error:

Admin | App Archive and Clean Up - Check Approvals and Archive

I get the following error:

Your flow could not turn on: Admin | App Archive and Clean Up - Check Approvals and Archive. Error: Flow client error returned with status code "Forbidden" and details "{"error":{"code":"DlpViolation","message":"Flow '' cannot be activated due to admin API policies."}}".

Our Office 365 API custom connector is listed in the Business Data Only group in the default DLP policy that's been applied to the CoE Environment on admin.powerapps.com. I've also tried making the Business Data Only group the default group but get the same error when attempting to turn on the flow. Our API connector uses the bearer token method instead of basic auth, could the error be related to this? Or is there some other app (currently in the No Business Data Allowed group) that the flow requires to work?

Regards,

Mel

**Comment by JeneferM-MSFT** [comment]
Can you validate, did you edit the flow to use a custom connector in place of one of the built in connectors used?

**Comment by melmgcoe** [comment]
Hello Jenefer,

Thanks for picking up this issue.

I did edit the flow as instructed in the CoE documentation.

But from what I remember, I only edited certain parts to use our CoE service account in terms of credentials to run the flow. I can't remember selecting any custom connectors.

Would it help to have a screen share to work through the flow itself? I considered sending you screenshots but the flow itself is quite large.

Regards,

[cid:image007.png@01D6291D.3E6A69C0]
[M&G Plc Logo]
[cid:image008.png@01D6291D.3E6A69C0]
Melvin Uy
Collaboration Technical Specialist
Workspace Collaboration O365
melvin.uy@uk.pgds.com <mailto:melvin.uy@uk.pgds.com>
t. 02039770030 (Internal: 60030)

10 Fenchurch Avenue
London EC3M 5AG
[cid:image009.png@01D6291D.3E6A69C0]


[cid:image009.png@01D6291D.3E6A69C0]
[cid:image009.png@01D6291D.3E6A69C0]
[cid:image011.png@01D6291D.3E6A69C0]

**Comment by JeneferM-MSFT** [comment]
I'm afraid I cannot do support calls for this, but I'm happy to work with you here. 
I'm still a little confused about what connectors are now in your flow, given your comments about custom API's and authentication methods. 

Can you please share a picture of what the connectors in this flow look like now? 
Browse to the flow details page > hit the edit button next to "Connections". Should look like this:
![image](https://user-images.githubusercontent.com/62659043/81828135-4e006f00-94ee-11ea-855f-0889a51dbb60.png)

**Comment by manuelap-msft** [comment]
To add to Jenefers comment, all the connectors listed in the screenshot above must be in the business data only group of your DLP policy.

Could it be that its SharePoint that's missing from your policy? That's a connector not used in any other Flows.

**Comment by melmgcoe** [comment]
Hi Manuela, Jenefer

Interestingly I don't see any connectors used for this particular flow:

[cid:image001.jpg@01D629F5.BD5EC7C0]

All the apps you mentioned including SharePoint exist in the data protection policy applied to the CoE environment.

Is it possible to edit the flow and manually add the missing connections?

Regards,

[cid:image008.png@01D629F5.BD5EC7C0]
[M&G Plc Logo]
[cid:image009.png@01D629F5.BD5EC7C0]
Melvin Uy
Collaboration Technical Specialist
Workspace Collaboration O365
melvin.uy@uk.pgds.com <mailto:melvin.uy@uk.pgds.com>
t. 02039770030 (Internal: 60030)

10 Fenchurch Avenue
London EC3M 5AG
[cid:image010.png@01D629F5.BD5EC7C0]


[cid:image010.png@01D629F5.BD5EC7C0]
[cid:image010.png@01D629F5.BD5EC7C0]
[cid:image012.png@01D629F5.BD5EC7C0]

**Comment by JeneferM-MSFT** [comment]
You pictures did not come through, please be sure to do your responses from GitHub in order to share images.

**Comment by melmgcoe** [comment]
I'm attaching the picture via Github as requested

![Flow capture](https://user-images.githubusercontent.com/62425530/81956831-b12ff580-9603-11ea-9c1f-ba04b53e1406.JPG)
.

**Comment by JeneferM-MSFT** [comment]
Can you confirm, did you go through the process of [activating the flows?](https://docs.microsoft.com/power-platform/guidance/coe/setup-core-components#activate-each-of-these-flows)

**Comment by melmgcoe** [comment]
Yes I did, and the flows were generated as part of the solution. I've tried saving a copy of the flow that has an issue and I now see the required connectors in the connection list. It seems odd that only this particular flow needs to be saved outside of the solution. When I try turning on the copy of the flow, I see this error:

![Flow capture](https://user-images.githubusercontent.com/62425530/82054388-c6199100-96b6-11ea-8f1b-d0f123d7f030.JPG)

Assume the connector called HTTP is required in our data policy and that this is a relatively new connector? All other connectors listed are in our data policy already.

**Comment by JeneferM-MSFT** [comment]
It looks like your DLP policy is blocking it. 

The environment must have a DLP policy that allows the following connectors to be used together: Common Data Service, Common Data Service (current environment), Office 365 Users, Power Platform for Admins, Power Automate Management, Office 365 Outlook, Power Apps for Admins, Power Apps for Makers, HTTP, and SharePoint. Those connectors must be in the business data–only bucket of the DLP policy for this environment.

Are all of those connectors in your business data only group? Do you have multiple DLP policies set up that might affect this environment?

**Comment by melmgcoe** [comment]
Thanks, I added HTTP as an allowed connector to our default data protection policy and the flow has now been enabled.

Is this a recently added connector by any chance? We track a list of connectors used in our default policy and didn't see this one when the list was exported via powershell.

@Power Platform CoE<mailto:PowerPlatformCoE@MandG.co.uk>, @Murray, Steven<mailto:Steve.Murray@Prudential.co.uk> - I've manually added an entry for this connector in our tracker spreadsheet.

Regards,

[cid:image001.png@01D62ACA.3DE49800]
[M&G Plc Logo]
[cid:image002.png@01D62ACA.3DE49800]
Melvin Uy
Collaboration Technical Specialist
Workspace Collaboration O365
melvin.uy@uk.pgds.com <mailto:melvin.uy@uk.pgds.com>
t. 02039770030 (Internal: 60030)

10 Fenchurch Avenue
London EC3M 5AG
[cid:image004.png@01D62ACA.3DE49800]


[cid:image004.png@01D62ACA.3DE49800]
[cid:image004.png@01D62ACA.3DE49800]
[cid:image006.png@01D62ACA.3DE49800]

**Comment by JeneferM-MSFT** [comment]
So glad you are unblocked! It isnt very new, here is [its announcement](https://flow.microsoft.com/blog/introducing-http-and-custom-connector-support-for-data-loss-prevention-policies/).  

However I did note that it isnt showing up in the standard places one would expect to see these in product documentation, so I am reaching out to see about fixing that. 

It was also missing from our own CoE documentation on the required DLP policy so we are fixing that now. Thanks for helping make CoE great!

**Comment by murray** [comment]
Hi. I'm not part of this project.

Do not include @Murray.

Thanks!
________________________________
